### PR TITLE
feature/CIV-1979/Display timeline as pdf

### PR DESCRIPTION
**Pull request by laperla11**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###
Display claimant timeline as pdf 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by laarmada — APPROVED** [review]

File: src/main/routes/features/response/timelineOfEvents/defendantTimelineController.ts [hunk 1]
```diff
@@ -13,18 +13,20 @@ import {TimeLineOfEvents} from '../../../../common/models/timelineOfEvents/timeL
...
+    form: form, theirTimeline: theirTimeline, pdfUrl,
+  });
 }
 
 defendantTimelineController.get(CITIZEN_TIMELINE_URL,
   async (req, res) => {
     try {
       const claim = await getCaseDataFromStore(req.params.id);
       const theirTimeline = claim.timelineOfEvents;
+      const pdfUrl = `/case/${req.params.id}/timeline/documents/${claim.extractDocumentId()}`;
```
Comment (taomoh) [inline src/main/routes/features/response/timelineOfEvents/defendantTimelineController.ts:27]: Maybe worth a constant?

**Review by taomoh — CHANGES_REQUESTED** [review]

File: config/custom-environment-variables.yaml [hunk 2]
```diff
@@ -13,7 +13,11 @@ services:
       host: REDIS_HOST
       port: REDIS_PORT
       tls: REDIS_TLS
-
+  dmStore:
+    clientID: IDAM_CLIENT_ID
```
[the remark this answers is not in the document]
Comment (laperla11) [reply config/custom-environment-variables.yaml]: changed the name as microserviceName.

File: src/main/app/client/serviceAuthProviderClient.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    return {
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    };
+  }
+
+  async getServiceAuthorisationToken(): Promise<string> {
+    const options = this.getConfig();
+    const serviceAuthTokenGeneratorUrl = '/testing-support/lease';
```
[the remark this answers is not in the document]
Comment (taomoh) [reply src/main/app/client/serviceAuthProviderClient.ts:28]: As discussed please raise a tech debt ticket to switch to the `/lease` endpoint in the future, which use one-time-password mechanism.
And then we can resolve this comment

File: src/main/routes/features/response/timelineOfEvents/defendantTimelineController.ts [hunk 4]
```diff
@@ -13,18 +17,20 @@ import {TimeLineOfEvents} from '../../../../common/models/timelineOfEvents/timeL
...
+    form: form, theirTimeline: theirTimeline, pdfUrl,
+  });
 }
 
 defendantTimelineController.get(CITIZEN_TIMELINE_URL,
   async (req, res) => {
     try {
       const claim = await getCaseDataFromStore(req.params.id);
       const theirTimeline = claim.timelineOfEvents;
+      const pdfUrl = CASE_TIMELINE_DOCUMENTS_URL.replace(':id', req.params.id);
```
Comment (taomoh) [inline src/main/routes/features/response/timelineOfEvents/defendantTimelineController.ts:31]: `documentId` substitution is missing

File: src/main/app/client/serviceAuthProviderClient.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
+import config from 'config';
+import Axios, {AxiosInstance, AxiosResponse} from 'axios';
+
+const {Logger} = require('@hmcts/nodejs-logging');
+const logger = Logger.getLogger('serviceAuthProviderClient');
+
+const xuiMicroserviceName = config.get<string>('services.dmStore.microserviceName');
+const serviceAuthTokenGeneratorUrl = config.get<string>('services.serviceAuthProvider.serviceAuthTokenGeneratorUrl');
+
```
Comment (osheaf01) [inline src/main/app/client/serviceAuthProviderClient.ts:9]: Just fmi, is this a typescript version of Java generics?

**Review by RosDiaf — APPROVED** [review]

File: src/test/unit/app/client/dmStoreClient.test.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+import axios, {AxiosInstance} from 'axios';
+import config from 'config';
+
+
+jest.mock('axios');
+const mockedAxios = axios as jest.Mocked<typeof axios>;
+const baseUrl: string = config.get('baseUrl');
+
+describe('DM Store Client', () => {
+  it('retrieve document by document id', async () => {
```
Comment (TatsyMaslova) [inline src/test/unit/app/client/dmStoreClient.test.ts:11]: should we put a test there for not so happy path when it throws the exception?

File: src/main/routes/features/response/timelineOfEvents/defendantTimelineController.ts (hunk 4 above)
Comment (taomoh) [reply src/main/routes/features/response/timelineOfEvents/defendantTimelineController.ts:31]: @laperla11 issue hasn't been resolved

File: src/main/routes/features/response/timelineOfEvents/defendantTimelineController.ts [hunk 7]
```diff
@@ -36,7 +42,8 @@ defendantTimelineController.post(CITIZEN_TIMELINE_URL, async (req, res) => {
     await form.validate();
     if (form.hasErrors()) {
       const claim = await getCaseDataFromStore(req.params.id);
-      renderView(form, claim.timelineOfEvents, res);
+      const pdfUrl = CASE_TIMELINE_DOCUMENTS_URL.replace(':id', req.params.id);
```
[the remark this answers is not in the document]
Comment (taomoh) [reply src/main/routes/features/response/timelineOfEvents/defendantTimelineController.ts:45]: @laperla11 issue hasn't been resolved

File: src/main/routes/features/response/timelineOfEvents/defendantTimelineController.ts (hunk 4 above)
Comment (laperla11) [reply src/main/routes/features/response/timelineOfEvents/defendantTimelineController.ts:31]: I have explained it personally

File: src/main/app/client/serviceAuthProviderClient.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (laperla11) [reply src/main/app/client/serviceAuthProviderClient.ts]: I created the tech debt ticket for it;
https://tools.hmcts.net/jira/browse/CIV-3046

[the remark this answers is not in the document]
Comment (taomoh) [reply src/main/app/client/serviceAuthProviderClient.ts]: test

**Review by taomoh — APPROVED** [review]

File: src/test/unit/app/client/dmStoreClient.test.ts (hunk 6 above)
Comment (laperla11) [reply src/test/unit/app/client/dmStoreClient.test.ts:12]: included the test for error exception.

**Pull request merged**